### PR TITLE
Add email-autocomplete w/ git auto-update via single package.json

### LINKED ISSUE #11512: [Request] Add Email-autocomplete
**Library name:** Email-autocomplete
**Git repository url:** https://github.com/10w042/email-autocomplete
**License:** MIT

=====================

For beginners who is interested in picking up this issue:

In this issue, we are going to host a new web front-end library on CDNJS, so that web developer can easily use it on our free CDN without additional download/upload process 🚀

If you'll like to work locally, I'll suggest you start pulling cdnjs repository with sparseCheckout: https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md, since there are so many files in this repo, without sparseCheckout, the whole process will be slow and will use more than 80GB disk space, using git command line can learn more skills and experience, if you don't like that, instead, you can also just do it on GitHub, it's more convenient, but it may be a little bit harder to fix problems on GitHub gui.

Here are the docs about how to add a library:
 - https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#d-adding-a-new-library-with-its-assets
 - https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#e-adding-a-new-library-by-a-single-packagejson

We'll also ask to setup auto-update config so that the added libraries can be updated to the latest version easily:
 - https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md

For beginners who is interested in join and help add this library, please feel free to tell me what kind of help you'll need, we'll help you send the valid pull request, thanks 😄

### PR DESCRIPTION
**Pull request by prathap0611**
close #11512, cc @10w042

Pull request for issue: #11512 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/10w042/email-autocomplete
 * Official website (optional, not the repository):
 * NPM package url (optional):
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 13
   - Count of stars: 199
   - Count of forks: 23
 * NPM download stats (optional):
   - Downloads in the last day: 
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/email-autocomplete/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "license": "MIT",
+  "author": "Low Yong Zhen <yz@stargate.io>",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/10w042/email-autocomplete.git",
+    "fileMap": [
+      {
+        "basePath": "dist",
+        "files": [
+          "**/*.js"
```
Comment (sufuf3) [inline ajax/libs/email-autocomplete/package.json]: Let's modify from `"**/*.js"` to `"**/*"`, because maybe this library add `.css` files under `dist` in future version.
And we usually use `**/*` which "basePath" is "dist".
Thank you.

Comment (prathap0611) [inline ajax/libs/email-autocomplete/package.json]: @sufuf3 Thank you for your comments. I have updated the code as per the comments.

**Comment by sufuf3** [comment]
@extend1994 How about in this comment? https://github.com/cdnjs/cdnjs/pull/11085#issuecomment-297126271
> If it at least links to its GitHub repo with a demo, I'll treat it as homepage

**Comment by extend1994** [comment]
@sufuf3 But there is no link to its repo.

**Comment by sufuf3** [comment]
@PeterDaveHello Could you give us a suggestion about homepage?
The Conditions of the establishment is one of https://github.com/cdnjs/cdnjs/pull/11085#issuecomment-297126271 and https://github.com/cdnjs/cdnjs/pull/10729#discussion_r104134423 is true?
As these two comments, I know that: 
1. if https://github.com/cdnjs/cdnjs/pull/10729#discussion_r104134423 = True, return true.
2. if https://github.com/cdnjs/cdnjs/pull/11085#issuecomment-297126271 = True, return true.

But, if https://github.com/cdnjs/cdnjs/pull/11085#issuecomment-297126271 is True but https://github.com/cdnjs/cdnjs/pull/10729#discussion_r104134423 is Fault, return what???

Although I think it fit https://github.com/cdnjs/cdnjs/pull/11085#issuecomment-297126271 so we can use https://10w042.github.io/email-autocomplete/demo/ as the homepage. 

Thank you.

**Comment by PeterDaveHello** [comment]
I don't know what are you asking ... This demo page has no repository info, isn't it that clear that we shouldn't use it as homepage?

File: ajax/libs/email-autocomplete/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "email-autocomplete",
+  "filename": "jquery.email-autocomplete.min.js",
+  "description": "jQuery plugin that displays in-place autocomplete suggestions for email input fields.",
+  "homepage": "https://10w042.github.io/email-autocomplete/demo/",
```
Comment (PeterDaveHello) [inline ajax/libs/email-autocomplete/package.json]: Please remove the "homepage" section as it's not really a homepage but just a demo page.

File: ajax/libs/email-autocomplete/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "fileMap": [
+      {
+        "basePath": "dist",
+        "files": [
+          "**/*"
+        ]
+      }
+    ]
+  }
+}
```
Comment (PeterDaveHello) [inline ajax/libs/email-autocomplete/package.json]: The last newline char seems to be missing?

**Comment by sufuf3** [comment]
@PeterDaveHello @prathap0611 Sorry for my mistake.

**Comment by PeterDaveHello** [comment]
@sufuf3 then you should help fix it, right?

**Comment by sufuf3** [comment]
Help remove the `homepage`

**Comment by prathap0611** [comment]
Sorry guys was bit busy last few days. @sufuf3 Thanks for your help. Will fix the new line char issue and push.

**Comment by PeterDaveHello** [comment]
@prathap0611 no worries at all, take your time :)

**Comment by sufuf3** [comment]
@cdnjs/library-reviewer 
Please help review this PR, Thank you.

**Comment by sufuf3** [comment]
@extend1994 not sure that if someone what to download v0.1.0 because he saw the [README](https://github.com/10w042/email-autocomplete/tree/v0.1.0#jqueryemail-autocompletejs-v010)

**Comment by extend1994** [comment]
@sufuf3 In my view, though we can see the text "jquery.email-autocomplete.js v0.1.0" in README.md, there's no file belongs to v0.1.0. So I think we should exclude this version.

**Comment by extend1994** [comment]
@prathap0611 We seem to need exclude v0.1.0, would you like to do it? Or we could do it for you 😃

**Comment by prathap0611** [comment]
@sufuf3 @extend1994 I have excluded v0.1.0 version. Thank you for your help.

File: .gitignore [hunk 4]
```diff
@@ -210,6 +210,7 @@ node_modules/
 /ajax/libs/slim.js/2.[0-5].*/
 /ajax/libs/europa/4.0.0-alpha*/
 /ajax/libs/inferno-compat/*-alpha.*/
+/ajax/libs/email-autocomplete/v0.1.0
```
Comment (extend1994) [inline .gitignore]: @prathap0611 Thanks! It's almost there! Let's modify this to `0.1.0`. And it would be good to use another commit to do the .gitignore part :)

**Comment by PeterDaveHello** [comment]
I won't do that to ignore v0.1.0 as it's a very small issue here.
